src/invoices/InvoiceDetails/hooks/useHasPendingOrders.js

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react';
+import { useOrders } from '../../../common/hooks';
+import { ORDER_STATUS } from '../constants';
+
+export const useHasPendingOrders = (orderLinesMap = {}) => {
+  const orderIds = useMemo(() => {
+    return Object.values(orderLinesMap)
+      .map(({ purchaseOrderId }) => purchaseOrderId);
+  },
+  [orderLinesMap]);
+
+  const {
+    isLoading,
+    orders,
+  } = useOrders(orderIds, { skip: !orderIds.length });
+
+  const hasPendingOrders = useMemo(() => {
+    if (!orders.length) {
+      return false;
```

Comment:
is it always `false` while loading? in case of big invoices it can be a problematic as action is enabled initially

Reply:
there is an initial condition `{ skip: !orderIds.length }` and in that case it is undefined.

Reply:
you didn't get my question, is `hasPendingOrders` false when we orders loading (orderIds is not empty) is in progress?

Reply:
Sorry, I didn't get your question. But what you are describing above it really depends. If there are orders it will skip the `if` statement and moves to the section `return orders.some(({ workflowStatus }) => workflowStatus === ORDER_STATUS.PENDING);`.

Reply:
from what I see, moment when orders are loading we have `orders.length === 0` and `hasPendingOrders` is false, is it correct? if so, I would propose to disable action until you exactly know that there are no pending orders instead of return false when orders are loading

try to connect 50 orders and you'll see that that action is enabled initially and becomes disabled after orders are fetched